File: sdk-python/nexus_sdk/adapters/llamaindex.py
"""
Pont LlamaIndex → Nexus.

    from nexus_sdk.adapters.llamaindex import NexusLlamaIndexAdapter

    agent = NexusLlamaIndexAdapter(
        index.as_query_engine(),
        name="base_documentaire",
        capabilities=["document_search", "rag"],
    )
    await agent.connect()

Un moteur de requête attend une chaîne : `input_key` vaut `"query"` par
défaut, de sorte qu'une tâche `{"query": "..."}` fonctionne directement.

Ce module n'importe pas LlamaIndex : voir `nexus_sdk.adapters`.
"""

from __future__ import annotations

from typing import Any, Iterable

from nexus_sdk.adapters.base import NexusAdapter


class NexusLlamaIndexAdapter(NexusAdapter):
    """Expose un moteur de requête ou de chat LlamaIndex sur le réseau Nexus."""

    def __init__(
        self,
        engine: Any,
        *,
        name: str,
        capabilities: Iterable[str] | None = None,
        input_key: str | None = "query",
        **kwargs,
    ):
        """
        Args:
            engine:    Résultat de `index.as_query_engine()`,
                       `as_chat_engine()`, ou un agent LlamaIndex.
            input_key: Clé extraite du dict de tâche, `"query"` par défaut.
        """
        super().__init__(
            engine, name=name, capabilities=capabilities,
            input_key=input_key, **kwargs,
        )

File: sdk-js/src/adapters/llamaindex.js
/**
 * Pont LlamaIndex.TS → InterMesh.
 *
 *   import { InterMeshLlamaIndexAdapter } from 'intermesh/adapters/llamaindex';
 *
 *   const agent = new InterMeshLlamaIndexAdapter(index.asQueryEngine(), {
 *     name: 'base_documentaire', capabilities: ['document_search', 'rag'],
 *   });
 *   await agent.connect();
 *
 * Un moteur de requête attend un objet `{ query: "..." }`, pas une chaîne
 * brute : contrairement à l'API Python (`query(str)`), LlamaIndex.TS
 * exige la forme objet. `inputKey` vaut `'query'` par défaut, de sorte
 * qu'une tâche InterMesh `{ query: "..." }` est extraite puis reconditionnée
 * dans la même forme avant l'appel. Pour un moteur de chat
 * (`chat({ message })`), passez `inputKey: 'message'`.
 *
 * Ce module n'importe pas `llamaindex` : voir `adapters/base.js`.
 */

import { InterMeshAdapter } from './base.js';

export class InterMeshLlamaIndexAdapter extends InterMeshAdapter {
  /**
   * @param {*} engine Résultat de `index.asQueryEngine()`,
   *   `index.asChatEngine()`, ou un agent LlamaIndex.TS.
   * @param {object} opts
   * @param {string} [opts.inputKey='query'] Clé extraite de l'objet de
   *   tâche, puis utilisée pour reconstruire l'objet attendu par le moteur.
   * @param {(data:*)=>*} [opts.inputAdapter] Transformation complète de
   *   l'entrée ; prime sur `inputKey` et son reconditionnement automatique.
   */
  constructor(engine, opts) {
    const { inputKey = 'query', inputAdapter, ...rest } = opts;
    const adapter = inputAdapter || ((data) => ({
      [inputKey]: (data !== null && typeof data === 'object') ? data[inputKey] : data,
    }));
    super(engine, { inputAdapter: adapter, ...rest });
  }
}
